From 327b59a574c81437f79b168655feff04b12ce56d Mon Sep 17 00:00:00 2001
From: William S Fulton <wsf@fultondesigns.co.uk>
Date: Sat, 28 Nov 2015 23:52:32 +0000
Subject: [PATCH] size_type correction for SwigPySequence_Cont

---
 Lib/python/pycontainer.swg | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Lib/python/pycontainer.swg b/Lib/python/pycontainer.swg
index 7c5cc37..ceb3b66 100644
--- a/Lib/python/pycontainer.swg
+++ b/Lib/python/pycontainer.swg
@@ -567,7 +567,7 @@ namespace swig
     typedef T value_type;
     typedef T* pointer;
     typedef int difference_type;
-    typedef int size_type;
+    typedef size_t size_type;
     typedef const pointer const_pointer;
     typedef SwigPySequence_InputIterator<T, reference> iterator;
     typedef SwigPySequence_InputIterator<T, const_reference> const_iterator;
