diff -up swig-3.0.12/configure.ac.go110 swig-3.0.12/configure.ac
--- swig-3.0.12/configure.ac.go110	2018-02-13 13:08:16.543632561 +0100
+++ swig-3.0.12/configure.ac	2018-02-13 13:56:20.241295155 +0100
@@ -2606,7 +2606,7 @@ else
     esac
     AC_MSG_CHECKING([whether go version is too old])
     case $go_version in
-    go1.1* | go1.0* | go1 )
+    go1.1 | go1.0* | go1 )
       AC_MSG_RESULT([yes - minimum version is 1.2])
       GO=
       GOOPT="-intgosize 32"
@@ -2624,7 +2624,7 @@ else
       ;;
     esac
     case $go_version in
-    go1.0* | go1 | go1.1*)
+    go1.0* | go1 | go1.1)
       GOOPT="$GOOPT -use-shlib"
       ;;
     go1.2*)
