diff -up swig-2.0.12/Examples/test-suite/guile/li_std_string_runme.scm.guile swig-2.0.12/Examples/test-suite/guile/li_std_string_runme.scm
--- swig-2.0.12/Examples/test-suite/guile/li_std_string_runme.scm.guile	2014-02-09 15:09:05.000000000 -0700
+++ swig-2.0.12/Examples/test-suite/guile/li_std_string_runme.scm	2014-03-01 08:33:41.343216202 -0700
@@ -5,5 +5,5 @@
 ; Note: when working with non-ascii strings in guile 2
 ;       locale must be set explicitly
 ;       The setlocale call below takes care of that
-(setlocale LC_ALL "")
+(setlocale LC_ALL "en_US.utf-8")
 (load "../schemerunme/li_std_string.scm")
