diff -up swig-1.3.38/Source/Modules/main.cxx.rh485540 swig-1.3.38/Source/Modules/main.cxx
--- swig-1.3.38/Source/Modules/main.cxx.rh485540	2009-02-16 16:27:07.000000000 +0100
+++ swig-1.3.38/Source/Modules/main.cxx	2009-02-16 16:29:05.000000000 +0100
@@ -954,7 +954,7 @@ int SWIG_main(int argc, char *argv[], La
     if (!s) {
       Printf(stderr, "Unable to locate '%s' in the SWIG library.\n", input_file);
     } else {
-      FILE *f = Swig_include_open(outfile);
+      FILE *f = Swig_open(outfile);
       if (f) {
 	fclose(f);
 	Printf(stderr, "File '%s' already exists. Checkout aborted.\n", outfile);
