diff -up swig-3.0.8/Lib/typemaps/fragments.swg.orig swig-3.0.8/Lib/typemaps/fragments.swg
--- swig-3.0.8/Lib/typemaps/fragments.swg.orig	2016-02-18 08:30:57.625989817 +0100
+++ swig-3.0.8/Lib/typemaps/fragments.swg	2016-02-18 08:31:13.260900411 +0100
@@ -168,7 +168,7 @@
 %fragment("SWIG_isfinite","header",fragment="<math.h>,<float.h>") %{
 /* Getting isfinite working pre C99 across multiple platforms is non-trivial. Users can provide SWIG_isfinite on older platforms. */
 #ifndef SWIG_isfinite
-# if defined(isfinite)
+# if defined(isfinite) || (defined __cplusplus && __cplusplus >= 201103L)
 #  define SWIG_isfinite(X) (isfinite(X))
 # elif defined(_MSC_VER)
 #  define SWIG_isfinite(X) (_finite(X))
