--- Source/Preprocessor/cpp.c.arch	2006-03-06 23:50:59.000000000 +0100
+++ Source/Preprocessor/cpp.c	2006-11-07 08:26:56.000000000 +0100
@@ -184,7 +184,11 @@
   included_files = NewHash();
 
   id_scan = NewSwigScanner();;
-
+  #ifdef __x86_64__
+    Preprocessor_define("__x86_64__", 0);
+  #elif defined __i386__
+    Preprocessor_define("__i386__", 0);
+  #endif
 }
 
 void Preprocessor_delete(void) {

