diff -up swig-2.0.7/Lib/python/pyhead.swg.rh830660 swig-2.0.7/Lib/python/pyhead.swg
--- swig-2.0.7/Lib/python/pyhead.swg.rh830660	2012-06-12 15:04:54.360670826 +0200
+++ swig-2.0.7/Lib/python/pyhead.swg	2012-06-12 15:05:24.119620754 +0200
@@ -5,6 +5,7 @@
 #define PyInt_Check(x) PyLong_Check(x)
 #define PyInt_AsLong(x) PyLong_AsLong(x)
 #define PyInt_FromLong(x) PyLong_FromLong(x)
+#define PyInt_FromSize_t(x) PyLong_FromSize_t(x)
 #define PyString_Check(name) PyBytes_Check(name)
 #define PyString_FromString(x) PyUnicode_FromString(x)
 #define PyString_Format(fmt, args)  PyUnicode_Format(fmt, args)
