From 604b3d009ce4ed170ebfcfb256fc5c91c08b1d0e Mon Sep 17 00:00:00 2001
From: William S Fulton <wsf@fultondesigns.co.uk>
Date: Sun, 13 Sep 2015 20:09:50 +0100
Subject: [PATCH] Ruby trackings bug fix support for 1.8

Issue #225
---
 Lib/ruby/rubytracking.swg | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/Lib/ruby/rubytracking.swg b/Lib/ruby/rubytracking.swg
index 37789c1..8f9f01b 100644
--- a/Lib/ruby/rubytracking.swg
+++ b/Lib/ruby/rubytracking.swg
@@ -11,6 +11,11 @@
 extern "C" {
 #endif
 
+#if !defined(ST_DATA_T_DEFINED)
+/* Needs to be explicitly included for Ruby 1.8 and earlier */
+#include <st.h>
+#endif
+
 /* Ruby 1.8 actually assumes the first case. */
 #if SIZEOF_VOIDP == SIZEOF_LONG
 #  define SWIG2NUM(v) LONG2NUM((unsigned long)v)
-- 
2.4.3

