diff -up swig-3.0.8/Examples/test-suite/li_boost_array.i.orig swig-3.0.8/Examples/test-suite/li_boost_array.i
--- swig-3.0.8/Examples/test-suite/li_boost_array.i.orig	2016-01-06 08:09:48.851786620 +0100
+++ swig-3.0.8/Examples/test-suite/li_boost_array.i	2016-01-06 08:10:54.028471052 +0100
@@ -31,7 +31,7 @@ namespace boost {
 
 %inline %{
 boost::array<int, 6> arrayOutVal() {
-  const char carray[] = { -2, -1, 0, 0, 1, 2 };
+  const signed char carray[] = { -2, -1, 0, 0, 1, 2 };
   boost::array<int, 6> myarray;
   for (size_t i=0; i<6; ++i) {
     myarray[i] = carray[i];
